persist/fs/retriever.go

```diff
@@ -437,9 +497,10 @@ func (req *retrieveRequest) resetForReuse() {
 	req.id = nil
 	req.start = time.Time{}
 	req.onRetrieve = nil
-	req.seekOffset = -1
+	req.indexEntry = IndexEntry{}
 	req.reader = nil
 	req.err = nil
+	req.doesNotExist = false
```

Comment:
nit: Super minor nit, perhaps `notFound` is more terse/Go-ish?

Reply:
good call, it should match the error name